src/client.rs

```diff
@@ -133,16 +123,18 @@ impl Tidal {
             url = [base_url, &url].concat().into();
         }
 
+        let Session { session_id, country_code, .. } = self.credentials.session.as_ref().unwrap();
+
         let mut headers = HeaderMap::new();
-        headers.insert("X-Tidal-SessionId", self.session_id().parse().unwrap());
+        headers.insert("X-Tidal-SessionId", session_id.parse().unwrap());
         headers.insert("Origin", "http://listen.tidal.com".parse().unwrap());
         if let Some(etag) = etag {
             headers.insert("If-None-Match", etag.parse().unwrap());
         }
 
         // Tidal's API requires countryCode to always be passed
         let mut query_params: HashMap<String, String> = HashMap::new();
-        query_params.insert("countryCode".to_owned(), self.country_code());
+        query_params.insert("countryCode".to_owned(), country_code.to_owned());
```

Comment:
Would it make sense to introduce an enum of query parameters here? Just so that you don't have hardcoded strings lying about the code.

Reply:
@jpalvarezl I will do that as part of a next PR (I will do it for the Headers too), let's try not to extend the scope of the PRs or it will keep growing and they will never be merged